runtime/lua/vim/treesitter/dev.lua

```diff
@@ -436,11 +457,150 @@ function M.inspect_tree(opts)
     buffer = buf,
     once = true,
     callback = function()
-      if api.nvim_win_is_valid(w) then
-        api.nvim_win_close(w, true)
+      close_win(w)
+    end,
+  })
+end
+
+-- TODO(MariaSolOs): Add helper functions for common code
+-- between the query editor and the tree inspector.
+--- @class EditQueryOpts
+--- @field bufnr integer? Buffer to draw the editor into. If omitted, a new
+---                       buffer is created.
+--- @field winid integer? Window id to display the editor buffer in. If omitted,
+---                       a new window is created with {command}.
+--- @field command string? Vimscript command to create the window. Default
+---                        value is "60vnew" or "new" if the inspector is open.
+---                        Only used when {winid} is nil.
+
+--- @private
+--- @param opts EditQueryOpts?
+M.edit_query = function(opts)
+  vim.validate({
+    opts = { opts, 't', true },
+  })
+
+  opts = opts or {}
+
+  local buf = api.nvim_get_current_buf()
+
+  -- Close any existing editor window
+  if vim.b[buf].dev_edit then
+    close_win(vim.b[buf].dev_edit)
+  end
+
+  local w = opts.winid
+  if not w then
+    local cmd = opts.command or '60vnew'
+    -- If the inspector is open, place the editor above it.
+    local base_buf = vim.b[buf].dev_base ---@type integer?
+    local inspect_buf = base_buf and vim.b[base_buf].dev_inspect
+    if base_buf and api.nvim_win_is_valid(inspect_buf) then
+      vim.api.nvim_set_current_win(inspect_buf)
+      buf = base_buf
+      cmd = 'new'
+    end
+    vim.cmd(cmd)
+    w = api.nvim_get_current_win()
+  end
+
+  local b = opts.bufnr
+  if b then
+    api.nvim_win_set_buf(w, b)
+  else
+    b = api.nvim_win_get_buf(w)
+  end
+
+  vim.b[buf].dev_edit = w
+  vim.bo[b].omnifunc = 'v:lua.vim.treesitter.query.omnifunc'
+  set_dev_properties(w, b)
+
+  local ok, parser = pcall(vim.treesitter.get_parser, buf)
+  if not ok then
+    return nil, 'No parser available for the given buffer'
+  end
+  local lang = parser:lang()
+
+  local ns = api.nvim_create_namespace('')
+
+  local function update_highlights()
+    api.nvim_buf_clear_namespace(buf, ns, 0, -1)
+    local query_content = table.concat(api.nvim_buf_get_lines(b, 0, -1, false), '\n')
+    local ok_query, query = pcall(vim.treesitter.query.parse, lang, query_content)
```

Comment:
`vim.F.npcall` could be used (query will be null in case of failure, `ok_query` no longer needed)

Reply:
It looks like a personal preference though. Would you like me to change line 94 as well then?

Reply:
Let's avoid using `vim.F` for now. It's future is tenuous, and there's little gain in this case.